designs/cloud_formation_api_gateway_support.md

```diff
@@ -0,0 +1,371 @@
+Cons: The customer may not be able to view their environments in different stages.
+
+*Approach: #2: *
+Create a dictionary of all the stages and display the url as localhost:3000/<Stage Name>/routeName
+Pros: This will allow people to view their code in different stages or environments.
+Cons: This will require refactoring parts of the SamApiProvider that handles the Api. Customers usually don't use different stages to test and it may not be worth the effort to test. 
+
+I recommend approach #1 as it follows the current standard and there is some data that very little people use the <Stage Name> when defining it. 
+
+**AWS::ApiGateway::Deployment **
+The deployment type defines which Apis objects should be available to the public. Normally, in the SAM template everything defined will have a deployment component, but the CloudFormation templates may have more bloat and information in them defining certain apis and deployments for Multi-stages.
+
+One approach is to support multiple stages with multiple deployments and the code will filter out the stages that are not deployed or defined. One problem with this approach is that the RestApiId, a characteristic of the RestApi, for a deployment is only localhost and a single endpoint and the current schema doesn't allow deploying on multiple ports and multiple local domains. This will bring in unneeded complexity such as requiring the customer defining N valid ports and N valid domains.
+
+Another approach, which I recommend, is to ignore the resource altogether as it always associated with a stage and to randomly pick a stage. 
+
+## **Updating the code**
+
+First, the Apis in AWS::ApiGateway::RestApi  will be parsed in the SAM Template in a similar way to the current AWS::Serverless:Api code. Stages and other Meta information in the Api will be parsed to support the CloudFormation template. The code will then be refactored to first translate the SAM templates inputted into the CloudFormation template using the SamTranslator. Once the Api code has been refactored, the function code can be refactored so that Serverless code is refactored out. 
+
+Some pseudo classes and function for the implementation 
+
+```python
+class InvokeContext(object):
+    ...
+    def __enter__(self):
+
+        # Grab template from file and create a provider
+        sam_template_data = self._get_template_data(self._template_file)
+        self._template_dict = self._get_clouformation_data(sam_template_data)
+        self._function_provider = CloudFormationFunctionProvider(self._template_dict,self.parameter_overrides
+    
+    # translate the template data to cloudformation template and return 
+    def _get_clouformation_data():
+        sam_translator = Translator(managed_policy_map=self.__managed_policy_map(),sam_parser=sam_parser, plugins=self.extra_plugins)
+        return sam_translator.translate(sam_template=template_copy, parameter_values=parameter_values)
+
+class CloudFormationFunctionProvider(SamBaseProvider):
```

Comment:
This doesn't parse the Method and resources right? That happens within the CfnProvider.

Can you update this section to match the implementation?